New midround antag: Demons of Sin

Marmio64 opened the pull request:
Adds a new kind of midround antag to mix it up. It borrows a little bit from devils but is also very different (and isn't meant to be some sort of remaster or replacement of devils). Basic rundown is that there are 5 demon types based on what I think are the most interesting sins to work with, and each has a flavor objective to complete and a couple powers to help them. These flavor objectives are pretty open ended and auto-complete so you can go about them anyway you want. Anyways, when created the demon arrives in the arrivals shuttle as an assistant and is given one of the sins.

Pride-
This demon's objective is to become as popular and influential to as many people as possible. They're unique in that this means they can actually help people since they're trying to become popular (as they're very prideful). Their unique ability is being able to summon a temporary lesser magic mirror at will, to make their image be greater, and they can offer it to others to try to become more popular. If you don't already know, a lesser magic mirror can change your name, appearance, and can change your race into any round start race such as humans, podpeople, lizards,etc.

Envy-
This demon's objective is to not let anyone become more popular then them. They also have a 50/50 chance of getting a second objective where they need to escape while disguised as someone else. To help with this their unique ability is Vanity Steal, which is basically the envy dagger from lavaland which lets them steal someone's exact look by hitting them.

Wrath-
This demon's objective is to cause as much pain as possible. Unlike the other demons this one is pretty much all murderbone, though killing people fast doesn't cause as much pain as doing it slowly, so hopefully they'll be a bit more interesting. They have a 50/50 chance of getting an assassination objective of someone. Their unique ability is Ignite, which can set someone on fire if they don't have antimagic protection. They also have more damage while in their True Demon Form, which will be touched on later.

Gluttony-
This demon has the goal of eating as much food as possible. Their objective is pretty harmless for the most part. Their unique ability is being able to make a forcefield out of gluttony walls, which only allow fat people to pass through. They can also gib and devour corpses while in their True Demon Form in order to eat them and heal.

Greed-
Last is Greed. A greed demon has to collect as much wealth as possible. Their unique ability is summoning a temporary greed slot machine which is exactly like the one on lavaland, allowing people to sacrifice max hp to play in order to have a chance to win a one use die of fate. They however cannot use this machine themselves, and are instead encouraged to 'auction' it off for cash to help with their goal.

<strike>Lust -
This demon has the objective of [REDACTED] and [DATA EXPUNGED]ing as many people as possible. Their unique ability is [REDACTED] which can █████████ and ████. </strike> This demon has been relocated to citadel station for 'reasons'.

Lastly, is the True Demon Form ability. All demons have this ability, though Wrath demons get more damage in this form and Gluttonous demons can eat corpses while in this form. While in this form they look similar to a red slaughter demon and pack a mean punch with 24 damage and 200 health. However, they are INCREDIBLY vulnerable to holy stuff in this form. Nullrods deal 40 damage to them and standing in the chapel drains 20 health every 2 seconds. Also demons can attack people to heal 10 hp each a hit, though this is cancelled if they're attacking anyone who is holy or antimagic.
![lesser demon](https://user-images.githubusercontent.com/60946370/155263502-18f0a532-7ec2-4de1-b416-17940497a728.JPG)

And finally be aware that sinful demons are very vulnerable to holy influence in their normal form too. They burn rapidly in the chapel (faster than vampires by a slight margin) and also take toxin damage from silver.

I believe that is all. Happy sinning?

# Changelog

:cl:  
rscadd: Adds a new midround antag: the demons of sin. TL;DR demons that have objectives according to their sin that can disguise among the crewmates or run around in demon form causing a ruckus.
/:cl:

Marmio64 committed:
demon spells, objectives, more

Marmio64 committed:
demon spells, objectives, more

Marmio64 committed:
finish up gluttony

Marmio64 committed:
finish up gluttony

Bibby0110 reviewed (dismissed):
cursory glance

Marmio64 committed:
Update code/modules/spells/spell_types/inflict_handler.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Marmio64 committed:
Update code/modules/spells/spell_types/inflict_handler.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Marmio64 committed:
Update code/modules/events/sinfuldemon.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Marmio64 committed:
Update code/modules/events/sinfuldemon.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Marmio64 committed:
Update code/modules/antagonists/demon/sins/pride.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Marmio64 committed:
Update code/modules/antagonists/demon/sins/pride.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Marmio64 committed:
Update code/modules/antagonists/demon/sins/pride.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Marmio64 committed:
Update code/modules/antagonists/demon/sins/pride.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Marmio64 committed:
Some spelling fixes from hopek

Co-authored-by: Hopekz <hopekxx@gmail.com>

Marmio64 committed:
Some spelling fixes from hopek

Co-authored-by: Hopekz <hopekxx@gmail.com>

Marmio64 committed:
Update code/modules/antagonists/demon/sins/gluttony.dm

Co-authored-by: Hopekz <hopekxx@gmail.com>

Marmio64 committed:
Update code/modules/antagonists/demon/sins/gluttony.dm

Co-authored-by: Hopekz <hopekxx@gmail.com>

Marmio64 committed:
is_helper and some documentation

Marmio64 committed:
is_helper and some documentation

Marmio64 committed:
Merge branch 'sinfuldemon' of https://github.com/Marmio64/Yogstation into sinfuldemon

Marmio64 committed:
objective fixes

Marmio64 committed:
objective fixes

Marmio64 committed:
greet sound, plus custom wrath and gluttony demon sprites

Marmio64 committed:
greet sound, plus custom wrath and gluttony demon sprites

Marmio64 committed:
lesser daemon icons

Marmio64 committed:
Merge branch 'sinfuldemon' of https://github.com/Marmio64/Yogstation into sinfuldemon

nesisknesis reviewed (commented):
This will not appear as the yogstation file is the one actually used
![image](https://user-images.githubusercontent.com/37936685/162639410-2b21264c-a0c9-4bd9-b86e-fa92520c4f3a.png)

pipe bomb sprite
![image](https://user-images.githubusercontent.com/37936685/162639459-01986265-dfda-4a9e-878e-54fc5ee80f70.png)

Marmio64 committed:
probably fixes the antag hud

Marmio64 committed:
greed slot machine has a better chance of winning

i'll see if this is too powerful, but hopefully the greed demons are making people spend money to use it regardless. Anyhow, even if you do happen to win this while spending your maximum life, you aren't even guaranteed to get something good, and there is a 10% chance to flatout die, so i'd say a higher chance of winning is in order.

Marmio64 committed:
hud

Marmio64 committed:
hud round 2

Marmio64 committed:
i dont know, another try

Marmio64 committed:
destroys the merge conflicts for now

Marmio64 committed:
Merge remote-tracking branch 'upstream/master' into sinfuldemon

Marmio64 committed:
centering
